### PR TITLE
Draft:Add interface and add typehinting

### PR DESCRIPTION
**Pull request by ColonelMoutarde**
Add the DatabaseDAOInterface

### CODE REVIEW

File: app/Models/DatabaseDAO.php [hunk 1]
```diff
@@ -3,7 +3,7 @@
 /**
  * This class is used to test database is well-constructed.
  */
-class FreshRSS_DatabaseDAO extends Minz_ModelPdo {
+class FreshRSS_DatabaseDAO extends Minz_ModelPdo implements DatabaseDAOInterface {
```
Comment (Alkarex) [inline app/Models/DatabaseDAO.php]: Why an interface when there is already a base class `FreshRSS_DatabaseDAO`?

File: app/Models/DatabaseDAOInterface.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	/**
+	 * @param string $table
+	 * @return array
+	 */
+	public function getSchema(string $table): array;
+
+	/**
+	 * @param array $dao
+	 * @return array
+	 */
```
Comment (Alkarex) [inline app/Models/DatabaseDAOInterface.php]: Comments can be removed when there is no additional information than the native type hints

**Comment by math-GH** [comment]
Is it a draft or an open PR?

**Comment by ColonelMoutarde** [comment]
A draft

**Review by Alkarex — APPROVED** [review]

**Comment by Alkarex** [comment]
I have simplified and finalised this PR.
For inspiration, more type hints welcome from the list of those compatible with PHP 7.0 https://php.net/language.types.declarations
In particular: `bool, float, int, string`
As contribution to https://github.com/FreshRSS/FreshRSS/issues/4112

**Pull request merged**